File: backend/Entities/TicketComment.cs
namespace backend.Entities;

public class TicketComment
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public string Comment { get; set; } = string.Empty;
    public Guid TicketId { get; set; }
    public Ticket Ticket { get; set; } = null!;

    public Guid UserId { get; set; }
    public User User { get; set; } = null!;
    public bool IsInternal { get; set; } = false;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? EditedAt { get; set; }
    public ICollection<TicketAttachment> Attachments { get; set; } = new List<TicketAttachment>();
}

File: frontend/types/ticket-comment.ts
import { TicketAttachmentDto } from "./ticket-attachment";
export interface TicketCommentDto{
    id : string;
    comment : string;
    createdById : string;
    createdBName : string;
    createdAt: string;
    editedAt : string | null;
    isEternal : string;
    attachments : TicketAttachmentDto;
}
export interface TicketCommentUpdateDto {
    comment : string;
    isInternal : string;
}
export interface TicketCommentCreateDto {
    comment : string;
    attachments : File[];
    isInternal : string;
}